src/presenced.cpp

```diff
@@ -350,6 +350,8 @@ Client::reconnect(const std::string& url)
                     mConnectPromise.reject(err.msg(), err.code(), WS_ERRTYPE_DNS);
                     mLoginPromise.reject(err.msg(), err.code(), WS_ERRTYPE_DNS);
                 }
+
+                return err;
```

Comment:
Remember - it's normal that a .fail() doesn't return the error, but handles it. Only in cases where you need to attach further handlers for that error is it useful to return the error. Similar to a rethrowing an exception from a `catch` block.